src/operator/tensor/sparse_retain-inl.h

```diff
@@ -267,7 +267,7 @@ void SparseRetainOpForwardRspImpl(mshadow::Stream<xpu> *s,
   if (!input_nd.storage_initialized()
       || idx_data.Size() == 0U
       || input_nd.shape()[0] == 0) {
-    FillZerosRspImpl(s, *output_nd);
```

Comment:
This was changed on purpose. Why revert this?

Reply:
It was a conflict and causing a compile error because my PR changes this function.  Why is it changed to const reference?  

When resolving the conflict I used the surrounding functions as a reference:

inline void FillDnsZerosRspImpl(mshadow::Stream<xpu> *s, NDArray *dst)
void PopulateFullIdxRspImpl(mshadow::Stream<xpu> *s, NDArray *dst)

btw why is there so much const reference being passed around when the object is then modified?  I see this everywhere.  Is this because lint complains anout not using a pointer for modified items rather than a non-const reference?

Reply:
NDArray is a pointer type. So its ok to modify the content of the object it points to.

Reply:
I think you mean NDArray::ptr_ is a const pointer type?

Reply:
ok, reverted